dart/dynamics/Skeleton.cpp

```diff
@@ -377,10 +377,22 @@ ConstSkeletonPtr Skeleton::getSkeleton() const
 
 //==============================================================================
 std::mutex& Skeleton::getMutex() const
+{
+  return getStdMutex();
+}
+
+//==============================================================================
+std::mutex& Skeleton::getStdMutex() const
 {
   return mMutex;
 }
 
+//==============================================================================
+std::unique_ptr<common::MutexReference> Skeleton::getCustomMutex() const
+{
+  return common::make_unique<common::SingleMutexReference>(mMutex);
```

Comment:
If this `Skeleton` is destructed, the `SingleMutexReference` returned by this function will contain a dangling reference to its mutex. `Skeleton`s are always owned by `shared_ptr`, we could prevent this by creating a new `OwnedMutexReference` class that stores a `std::mutex&` and a `std::shared_ptr<void>`. We would construct it here by passing in `mMutex` and `shared_from_this()`.

This would prevent the `Skeleton` from being destructed until all `OwnedMutexReference`s go out of scope. As an interesting alternative, we only prevent the `Skeleton` from being destructed while its lock is acquired by replacing the `std::shared_ptr<void>` with an `std::weak_ptr<void>`. Every function on `MutexRefernece` would `lock()` the `weak_ptr` and skip the lock if the `Skeleton` was already destructed. It is presumably safe to skip the lock if all of the data that it is protecting has already been deleted. 😉 

(I am not sure if the `weak_ptr` approach is necessary or desirable. I just thought mentioned it because I thought it was interesting.)

Reply:
I took the alternative suggestion (i.e., holding the skeleton's pointer in `std::weak_ptr`) without introducing a new class.

I'm still open to adding `OwnedLockableReference` though.